File: design-pattern-go/src/patterns/structural/facade/ThirdpartyLibraries.go
package facade

import (
	"strings"
)

type VideoFile struct {
	Filename string
}

func NewVideoFile(filename string) VideoFile {
	return VideoFile{Filename: filename}
}

type CompressionCodec struct{}
type OggCompressionCodec struct {
	CompressionCodec
}

func NewOggCompressionCodec() OggCompressionCodec {
	return OggCompressionCodec{}
}

type MPEG4CompressionCodec struct {
	CompressionCodec
}

func NewMPEG4CompressionCodec() MPEG4CompressionCodec {
	return MPEG4CompressionCodec{}
}

func CodeFactoryExtract(file VideoFile) string {
	return strings.Join([]string{
		"CodeFactory.extract",
		file.Filename,
	}, "")
}

func BitrateReaderRead(value string, source string) string {
	return strings.Join([]string{
		"BitrateReader.read: ",
		value,
		",",
		source,
	}, "")
}
func BitrateReaderBuffer(value string, source CompressionCodec) string {
	return strings.Join([]string{
		"BitrateReader.buffer: ",
		value,
	}, "")
}

type AudioMaker struct{}

func NewAudioMaker() AudioMaker {
	return AudioMaker{}
}

func (a *AudioMaker) Fix(value string) string {
	return strings.Join([]string{"AudioMixer.Fix: ", value}, "")
}

File: design-pattern-python/patterns/structural/facade/third_party_libraries.py
from isort import file


class VideoFile:
    def __init__(self, filename: str) -> None:
        self.filename = filename


class CompressionCodec:
    def __init__(self) -> None:
        pass


class OggCompressionCodec(CompressionCodec):
    def __init__(self) -> None:
        super().__init__()


class MPEG4CompressionCodec(CompressionCodec):
    def __init__(self) -> None:
        super().__init__()


def CodecFactory_extract(file: VideoFile) -> str:
    return file.filename


def BitrateReader_read(value: str, source: str) -> str:
    return f"BitrateReader.read: {value}, {source}"


def BitrateReader_buffer(value: str, compression: CompressionCodec) -> str:
    return f"BitrateReader.buffer: {value}"


class AudioMixer:
    def fix(self, value: str) -> str:
        return f"AudioMixer.fix: {value}"
